### DISCUSSION TITLE
How to run Exe with parameters?

### DISCUSSION THREAD

**Discussion by ghost1372** [microsoft/WindowsAppSDK#1155] [Q&A] [answered]
In UWP we can run Exe with the following method
```
<Extensions>
  <desktop:Extension xmlns:desktop="http://schemas.microsoft.com/appx/manifest/desktop/windows10"
				   Category="windows.fullTrustProcess"
				   Executable="Assets\UnRAR.exe" />
</Extensions>
```
and
`await FullTrustProcessLauncher.LaunchFullTrustProcessForCurrentAppAsync();`

But this method does not seem to work, i get this error

`System.Runtime.InteropServices.COMException: 'Call context cannot be accessed after call completed. (0x80010117)'
` 
![1](https://user-images.githubusercontent.com/9213496/127745206-45e05023-88e7-4a37-a988-5efad86e9feb.png)

**Comment by btueffers** [comment]
It appears that API is attempting to run it's own process as admin, which is currently only supported on Windows 11 #896

**Reply by ghost1372** [reply]
Is it possible to use it on Windows 10 in the future?

**Reply by andrewleader** [reply]
Right now we don't have any current plans to support elevated apps using WinAppSDK on Windows 10. If that's something you need, I can log your feature request for that. We're always evaluating options, however it's seeming like it'll be really costly for us to attempt to support running elevated on Windows 10.

**Reply by dhoehna** [reply]
@andrewleader unless I can't see it, how is this a windowsAppSDK thing?  Isn't FullTrustProcessLauncher already on windows 10?  To allow UWP apps to launch full trust processes?

**Reply by andrewleader** [reply]
@dhoehna ghost1372 has a WinUI 3 app (which uses WinAppSDK), and they're trying to launch their own app as full trust (so trying to launch their WinUI 3 app as full trust).

Running a WinUI 3 app or app that uses WinAppSDK as full trust isn't supported on Windows 10 (only on Windows 11).

**Reply by dhoehna** [reply]
Yep.  Didn't notice that.  Thank you for pointing that out.

**Reply by riverar** [reply]
@andrewleader Wait, what? That should be supported. I think you mean *elevated* full trust apps are not supported, right?

**Reply by andrewleader** [reply]
Yes I mean elevated/admin, sorry for the confusion

**Comment by andrewleader** [comment, marked as answer]
Actually I read this wrong... **so the scenario is you want to distribute an EXE alongside your WinUI 3 MSIX desktop app, and you want to launch that EXE dynamically from your code as full trust.**

Since a WinUI 3 app is a desktop app, you can use the `Process.Start` API. Here's a full example of launching the VS installer, distributed within the app itself, requesting it to run elevated. You don't need to specify any `<desktop:Extension>` in your manifest either. You should be able to pass arguments via `Process.Start` too. And this works on Windows 10 😊

![image](https://user-images.githubusercontent.com/13246069/128776398-e1838b9e-f755-497c-89d9-99fe50561ec6.png)

The thing that doesn't work on Windows 10 (only Windows 11) is running your own WinUI 3 app as elevated/admin. But you can request another process to launch as admin 😊
